src/Core/Managed/Shared/Extensibility/Implementation/JsonSerializer.cs

```diff
@@ -54,6 +54,51 @@ public static byte[] Serialize(IEnumerable<ITelemetry> telemetryItems, bool comp
         }
 
         /// <summary>
+        /// Converts serialized telemetry items to a byte array.
+        /// </summary>
+        /// <param name="telemetryItems">Serialized telemetry items.</param>
+        /// <param name="compress">Should serialization also perform compression.</param>
+        /// <returns>The compressed and serialized telemetry items.</returns>
+        [SuppressMessage("Microsoft.Usage", "CA2202:Do not dispose objects multiple times", Justification = "Disposing a MemoryStream multiple times is harmless.")]
+        public static byte[] ConvertToByteArray(string telemetryItems, bool compress = true)
+        {
+            if (string.IsNullOrEmpty(telemetryItems))
+            {
+                throw new ArgumentNullException("telemetryItems");
+            }
+
+            var memoryStream = new MemoryStream();
+            using (Stream compressedStream = compress ? CreateCompressedStream(memoryStream) : memoryStream)
+            using (var streamWriter = new StreamWriter(compressedStream, TransmissionEncoding))
+            {
+                streamWriter.Write(telemetryItems);
+            }
+
+            return memoryStream.ToArray();
+        }
+
+        /// <summary>
+        /// Deserializes and decompress the telemetry items into a JSON string.
+        /// </summary>
+        /// <param name="telemetryItemsData">Serialized telemetry items.</param>
+        /// <param name="compress">Should deserialization also perform decompression.</param>
+        /// <returns>Telemetry items serialized as a string.</returns>
+        public static string Deserialize(byte[] telemetryItemsData, bool compress = true)
+        {
+            var memoryStream = new MemoryStream(telemetryItemsData);
+            
+            using (Stream decompressedStream = new GZipStream(memoryStream, CompressionMode.Decompress))
+            {
```

Comment:
I need Decompress, not Compress parameter that is used in that method. I would actually remove this helper method. As I understand we had it because we were supporting not compressed streams before which is not the case any more.